tests/41end-to-end-keys/06-device-lists.pl

```diff
@@ -208,9 +208,12 @@ sub sync_until_user_in_device_list_id
          my $alice_device_keys = $alice_keys->{ $user2->device_id };
 
          # TODO: Check that the content matches what we uploaded.
-
-         assert_eq( $alice_device_keys->{"unsigned"}->{"device_display_name"},
-                    "test display name" );
+         # Device display names aren't mandated in the POST /user/keys/query response,
+         # and they're considered optional in the GET /user/devices/{userId} response.
+         # So accept either a match or a lack of key.
+         my $device_display_name = $alice_device_keys->{"unsigned"}->{"device_display_name"};
+         $device_display_name = undef or $device_display_name = "test display name" or
```

Comment:
wtf was I smoking---this `=` is assinging, not testing (`==`?)

I'm going to have to read perl docs now, aren't I?